rpc_jobs/setup_infra.yml

```diff
@@ -0,0 +1,137 @@
+                  "install",
+                  "-r ${{env.WORKSPACE}}/rpc-gating/{name_lower}/role_requirements.yml",
+                  "-p ${{env.WORKSPACE}}/roles"
+                )
+              }}
+            }} // stage
+
+            stage("Prepare Environment"){{
+
+              // write the clouds.yaml file
+              env.OS_CLIENT_CONFIG_FILE = common.writeCloudsCfg()
+
+              // Tell ansible where to find roles
+              env.ANSIBLE_ROLES_PATH = "${{env.WORKSPACE}}/roles"
+
+              // Tell ansible where the dynamic inventory is
+              env.ANSIBLE_INVENTORY = "${{env.WORKSPACE}}/rpc-gating/scripts/ansible_v2_3_2_0_1_contrib_inventory_openstack.py"
+
+              // Tell the dynamic inventory script to only use the 'public_cloud' cloud
+              env.OS_CLOUD = "public_cloud"
+
+              // Tell ansible where the ssh private key is
+              env.ANSIBLE_PRIVATE_KEY_FILE = "${{env.JENKINS_SSH_PRIVKEY}}"
+
+            }} // stage
+
+            stage("Prepare Instances"){{
+              dir('rpc-gating/playbooks'){{
+
+                // Run the setup playbook
+                common.venvPlaybook(
+                  playbooks: [
+                    "setup_openstack_instances.yml"
+                  ],
+                  args: [
+                    "-v",
+                    "-e '@${{env.WORKSPACE}}/rpc-gating/{name_lower}/instance_list.yml'"
+                  ]
+                ) //venvPlaybook
+
+                // We need an inventory refresh, so we execute
+                // this using a fresh shell instead of including
+                // the playbooks in the previous ansible-playbook
+                // play list.
+
+                // Run the instance prep plays
+                common.venvPlaybook(
+                  playbooks: [
+                    "drop_ssh_auth_keys.yml",
+                    "slave_security.yml"
+                  ],
+                  args: [
+                    "-v",
+                    "-e target_hosts={name_lower}_server"
+                  ]
+                ) //venvPlaybook
+
+              }} //dir
+            }} // stage
+
+            stage("Setup {name}"){{
+              dir('rpc-gating/{name_lower}'){{
+
+                // Run the setup playbook
+                common.venvPlaybook(
+                  playbooks: [
+                    "setup_{name_lower}.yml"
+                  ],
+                  args: [
+                    "-v"
+                  ]
+                ) //venvPlaybook
+              }} //dir
+            }} // stage
+          }} //withCredentials
+        }} catch (e) {{
+          print(e)
+          throw(e)
+        }} finally {{
+          common.archive_artifacts()
+        }}
+      }} // globalWraps
+
+- project:
+    name: 'Setup-Infra'
+    repo_name: "rpc-gating"
+    repo_url: "https://github.com/rcbops/rpc-gating"
+    branch: "master"
+    jira_project_key: "RE"
+    CRON: "@daily"
```

Comment:
Could you use `H H * * 2-7` instead of @daily? This will avoid a conflict with #1040, see https://rpc-openstack.atlassian.net/wiki/spaces/RE/pages/469794817/RE+Infrastructure+Maintenance+Window for more info.

Reply:
Actually don't worry about changing this, we'll see which merges first then the other can rebase. Then you can use `{CRON_DAILY}` which won't exist until #1040 merges.